### PR TITLE
Properly append preload / includes args on Merger

### PR DESCRIPTION
**Pull request by huoxito**
Noticed this one while looking at this example:

``` ruby
class Product < ActiveRecord::Base
  has_many :variants
  has_many :translations
end

class Translation < ActiveRecord::Base
  belongs_to :product
end

class Variant < ActiveRecord::Base
  belongs_to :product
end

class BugTest < Minitest::Test
  def test_merge_stuff
    product = Product.create! name: 'huhu'
    variant = Variant.create! product_id: product.id
    Translation.create! locale: 'en', product_id: product.id

    product_relation = Product.all
                              .preload(:translations)
                              .joins(:translations)
                              .merge(Translation.where(locale: 'en'))
                              .where(name: 'huhu')

    assert_equal variant, Variant.joins(:product).merge(product_relation).first
  end
end
```

It doesn't seem right to `preload` the translations on the Variant since
it was never called there. No idea if this is the best solution either
but can't think of why one would need to `preload!` again when merging
relations. After the query gets executed and AR actually tries to load
the associations it will error out with something like this:

> ActiveRecord::AssociationNotFoundError: Association named 'translations' was not found on Variant; perhaps you misspelled it?

BTW this prevents the error above from happening but it doesn't fix the issue since the preload will never happen. I found this commit https://github.com/rails/rails/commit/dc764fcc348d562376add26ff8ef5173946b575b and think this could be a similar scenario. Assuming this is the way to go I will try looking further later. And I see there are [some other PRs](ttps://github.com/rails/rails/search?p=1&q=preload&state=open&type=Issues&utf8=%E2%9C%93) involving preload issues .. perhaps some of them might help address this as well. Anyway just figured starting with a PR wouldn't do any harm.

Let me know it this doesn't make sense pls cheers

### CODE REVIEW

**Comment by huoxito** [comment]
just amended the previous commit to add a possible fix for the issue, pls take a look and advise when you have a chance

File: activerecord/lib/active_record/relation/merger.rb [hunk 1]
```diff
@@ -82,6 +83,27 @@ def merge
 
       private
 
+      def merge_preloads
+        return if other.preload_values.blank? && other.includes_values.blank?
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/relation/merger.rb]: can we use `empty?` here?

Comment (huoxito) [inline activerecord/lib/active_record/relation/merger.rb]: yep I think so build is still green with `empty?`, amended the commit

**Comment by huoxito** [comment]
something seems wrong with triggering travisci builds, just push forced and it triggered the build for my previous commit https://travis-ci.org/rails/rails/builds/64362612 but not for the one I just pushed

**Comment by rafaelfranca** [comment]
Yeah, Travis is having a hard time to catch all my commits. Seems I'm faster than it :rabbit: :runner:

**Comment by huoxito** [comment]
@rafaelfranca any chance this would get into 4-2-stable? first noticed the issue on rails 4.2.1, there were some conflicts so I submitted another PR (see above ^^) just in case it's fine

**Pull request merged**